dev-client/src/model/soilIdMatch/actions/soilIdMatchActions.ts

```diff
@@ -22,34 +22,57 @@ import {ThunkAPI} from 'terraso-client-shared/store/utils';
 import {Coords} from 'terraso-client-shared/types';
 
 import {
-  siteEntryForMatches,
+  ClientSoilIdFailureReason,
+  siteEntry,
   siteEntryForStatus,
   SoilIdEntry,
-  tempLocationEntryForMatches,
-  tempLocationEntryForStatus,
+  tempLocationEntry,
 } from 'terraso-mobile-client/model/soilIdMatch/soilIdMatches';
-import {AppState} from 'terraso-mobile-client/store';
+import {
+  updateSiteMatchesAfterTimeout,
+  updateTempMatchesAfterTimeout,
+} from 'terraso-mobile-client/model/soilIdMatch/soilIdMatchSlice';
+import {AppDispatch, AppState} from 'terraso-mobile-client/store';
+
+// TODO-cknipe: Should this live here or elsewhere? Also, change back to 20000
+const TIMEOUT_MS = 20000;
 
 export const fetchTempLocationBasedSoilMatchesThunk = async (
   coords: Coords,
   _: User | null,
-  __: ThunkAPI,
-) => fetchTempLocationBasedSoilMatches(coords);
+  {dispatch}: ThunkAPI,
+) => fetchTempLocationBasedSoilMatches(coords, dispatch);
+
+const timeoutError = {
+  __typename: 'SoilIdFailure' as const,
+  reason: 'TIMEOUT' as ClientSoilIdFailureReason,
+};
+type TimeoutError = typeof timeoutError;
```

Comment:
commnet: it feels a bit funny to be simulating GraphQL output here, instead of directly giving the error in the format we eventually convert the GraphQL response to